src/lib/optimizer/chunk_statistics.cpp

```diff
@@ -0,0 +1,77 @@
+#include "chunk_statistics.hpp"
+
+#include <iterator>
+
+#include "all_parameter_variant.hpp"
+#include "resolve_type.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+template <typename T>
+static std::shared_ptr<ChunkColumnStatistics> build_statistics_from_dictionary(const pmr_vector<T>& dictionary) {
+  auto statistics = std::make_shared<ChunkColumnStatistics>();
+  // only create statistics when the compressed dictionary is not empty
+  if (!dictionary.empty()) {
+    auto min_max_filter = std::make_unique<MinMaxFilter<T>>(dictionary.front(), dictionary.back());
+    statistics->add_filter(std::move(min_max_filter));
+
+    // no range filter for strings
+    if
+      constexpr(std::is_arithmetic_v<T>) {
```

Comment:
please add `// clang-format off/on` so that `if` and constexpr` stay on the same line.